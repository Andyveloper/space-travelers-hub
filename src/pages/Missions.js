const Missions = () => (
  <div><h1>Hola Mision</h1></div>
);

export default Missions;
