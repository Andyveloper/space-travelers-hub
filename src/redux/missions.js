import { createSlice } from '@reduxjs/toolkit';

const initialState = {
  missions: [],
  loading: false,
};

const reducer = createSlice({
  name: 'missions',
  initialState,
  reducers: {
    setMissions: (state, action) => ({
      ...state,
      missions: action.payload,

    }),
  },
});

export const { setMissions } = reducer.actions;
export default reducer;
