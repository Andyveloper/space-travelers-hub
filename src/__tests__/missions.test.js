import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import getMissions from '../redux/apiMissions';
import missions from '../redux/missions';
import Missions from '../pages/Missions';

const renderJoinedMissions = (joined) => {
  const { tree } = render(
    <Provider store={store}>
      <Missions
        joined={joined}
        id={1}
        name="Thaicom"
        description="Test description"
      />
    </Provider>,
  );
  return tree;
};

describe('Test missions button and message', () => {
  it('Renders first Mission', () => {
    expect(renderJoinedMissions(true)).toMatchSnapshot();
  });
});

describe('Test missions reducers and functions', () => {
  it('Fetches data from API', async () => {
    await store.dispatch(getMissions());
    expect(store.getState().missions).toBeTruthy();
  });

  it('Returns the initial state', () => {
    expect(missions.reducer(undefined, { type: undefined })).toEqual(
      {
        missions: [],
        status: null,
      },
    );
  });
});
