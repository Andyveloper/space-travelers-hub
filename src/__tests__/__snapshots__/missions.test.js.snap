// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test missions button and message Renders first Mission 1`] = `undefined`;
